wasabi/tables.py

```diff
@@ -1,54 +1,56 @@
-# coding: utf8
-from __future__ import unicode_literals, print_function
 import os
+from itertools import zip_longest
+from typing import Collection, Dict, Iterable, List, Optional, Sequence, Union
+from typing import cast
+
+from typing_extensions import Literal
 
 from .util import COLORS
 from .util import color as _color
-from .util import supports_ansi, to_string, zip_longest, basestring_
-
+from .util import supports_ansi
 
 ALIGN_MAP = {"l": "<", "r": ">", "c": "^"}
 
 
 def table(
-    data,
-    header=None,
-    footer=None,
-    divider=False,
-    widths="auto",
-    max_col=30,
-    spacing=3,
-    aligns=None,
-    multiline=False,
-    env_prefix="WASABI",
-    color_values=None,
-    fg_colors=None,
-    bg_colors=None,
-):
+    data: Union[Collection, Dict],
+    header: Optional[Iterable] = None,
+    footer: Optional[Iterable] = None,
+    divider: bool = False,
+    widths: Union[Iterable[int], Literal["auto"]] = "auto",
+    max_col: int = 30,
+    spacing: int = 3,
+    aligns: Optional[Union[Iterable[Literal["r", "c", "l"]], str]] = None,
+    multiline: bool = False,
+    env_prefix: str = "WASABI",
+    color_values: Optional[Dict] = None,
+    fg_colors: Optional[Iterable] = None,
+    bg_colors: Optional[Iterable] = None,
+) -> str:
     """Format tabular data.
 
-    data (iterable / dict): The data to render. Either a list of lists (one per
+    data (Union[Collection, Dict]): The data to render. Either a list of lists (one per
         row) or a dict for two-column tables.
-    header (iterable): The header columns.
-    footer (iterable): The footer columns.
+    header (Optional[Iterable]): Optional header columns.
+    footer (Optional[Iterable]): Optional footer columns.
     divider (bool): Show a divider line between header/footer and body.
-    widths (iterable or 'auto'): Column widths in order. If "auto", widths
+    widths (Union[Iterable[int], Literal['auto']]): Column widths in order. If "auto", widths
         will be calculated automatically based on the largest value.
     max_col (int): Maximum column width.
     spacing (int): Spacing between columns, in spaces.
-    aligns (iterable / unicode): Column alignments in order. 'l' (left,
-        default), 'r' (right) or 'c' (center). If a string, value is used
-        for all columns.
+    aligns (Optional[Union[Iterable[Literal['r', 'c', 'l']], str]]): Optional column alignments
```

Comment:
For the docstring, maybe just?

```suggestion
    aligns (Optional[Union[Iterable[str], str]]): Optional column alignments
```